### PR TITLE
 Migrate ClusterTPAssociationHeterogeneous to the new framework 

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
Split #324, part ? of N.

File: SimTracker/TrackerHitAssociation/plugins/ClusterTPAssociationHeterogeneous.cc [hunk 1]
```diff
@@ -44,9 +45,8 @@
 
 #include "ClusterSLOnGPU.h"
 
-class ClusterTPAssociationHeterogeneous : public HeterogeneousEDProducer<heterogeneous::HeterogeneousDevices<
-          heterogeneous::GPUCuda, heterogeneous::CPU>>
-{
+class ClusterTPAssociationHeterogeneous
+    : public HeterogeneousEDProducer<heterogeneous::HeterogeneousDevices<heterogeneous::GPUCuda, heterogeneous::CPU>> {
```
Comment (makortel) [inline SimTracker/TrackerHitAssociation/plugins/ClusterTPAssociationHeterogeneous.cc]: `ClusterTPAssociationHeterogeneous` seems to stay `HeterogeneousEDProducer`, so the title of this PR is misleading.

By cursory look this PR introduces the `TrackingRecHit2DCUDA`, and migrates the `CklusterTPAssociationHeterogeneous` do use that. The hit-producing code `SiPixelRecHitHeterogeneous` appears to still produce `HeterogeneousProduct`, so there is nothing that would produce `TrackingRecHit2DCUDA`.

Should this PR also include the migration of `SiPixelRecHitHeterogeneous` to the new framework?

**Comment by fwyzard** [comment]
@makortel this PR is actually supposed to be one of the last steps if #338 ; does it look better in that context ?

**Comment by makortel** [comment]
> this PR is actually supposed to be one of the last steps if #338 ; does it look better in that context ?

Sort of yes, but I just assumed that this PR would have been self-contained as it was split from #324/#338 (as all the other similar PRs were, right?).

**Pull request merged**